plugins.json

```diff
@@ -9,5 +9,10 @@
       "version": "1.0.5"
     }
   },
-  "plugins": [ "auto-sizes" ]
+  "plugins": {
+    "auto-sizes": {
+      "slug": "auto-sizes",
+      "version": "1.0.1"
```

Comment:
Should this be necessary if the `version` is already present here:

https://github.com/WordPress/performance/blob/36fe0dce276a81db16a7bd0dc13c453d0d207314/plugins/auto-sizes/auto-sizes.php#L8

Should this not rather parse the version out of the plugin file? Or we also have it present in the `readme.txt`:

https://github.com/WordPress/performance/blob/36fe0dce276a81db16a7bd0dc13c453d0d207314/plugins/auto-sizes/readme.txt#L7

I think we should avoid having this version in three different places.

Reply:
Or the version could be stored in `plugins.json` like you have here, but if so, then `n.e.x.t` should be the version in `auto-sizes.php` and `readme.txt` to avoid having to manually keep them in sync.

Reply:
Like the idea of storing the version in a single place only. As mentioned `n.e.x.t` strategy can be used to populate the version on the fly.

Reply:
Thanks, @westonruter, for bringing this up. The plugin versions were added for the deploy process per https://github.com/WordPress/performance/issues/935#issuecomment-1952027981. Since there is no build project for `plugins`, in my opinion, we should discuss this first and address it in a follow-up issue. What do you think? cc. @joemcgill

Reply:
Yes, I saw that comment. Still, not ideal to have the version in triplicate. If you want to address that in another issue, sure.

Reply:
Open a separate issue for further work: #1006

Reply:
I spent some time looking into how we could modify the release job to avoid the duplication of version numbers here for this, and I'm not sure it's worth the effort for now. Once we finish migrating all of the standalone modules to the plugins folder we can clean this up more easily.

Reply:
@mukeshpanchal27 @joemcgill I just voiced a similar concern as @westonruter in https://github.com/WordPress/performance/issues/935#issuecomment-1965535471, and his feedback is precisely why the requirements only mentioned plugin slugs (no need to duplicate the version number).

I'm okay for this to be addressed in a separate issue, but since this PR is actually making the change and the `feature/modules-to-plugins` branch currently uses the _correct_ format, it's a little strange IMO to go ahead with this change just to change it back afterwards.

If this PR is blocking other work and therefore would benefit from a merge ASAP, I'm okay to fix this again in a follow up issue, but it should be addressed shortly after this PR, in time for 3.0, as it requires changes to the `plugins.json` file format/spec, which will affect documentation.

Reply:
Thanks for the feedback.

In https://github.com/WordPress/performance/pull/1000/commits/a4cf3bf058093370ccdf067027edd53049493922, I have reverted the version-related changes for plugins and instead fetched the version from the `readme.txt` file. I updated the script accordingly. I opted not to use the plugin's main file because different plugins have different main files, such as `plugin/load.php` or `plugin/plugin.php`.